frontend/src/app/user/duck-list/duck-list.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Duck } from '@app/_models';
+import { DuckService } from '@app/_services/duck.service';
+import { map, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-duck-list',
+  templateUrl: './duck-list.component.html',
+  styleUrls: ['./duck-list.component.scss']
+})
+export class DuckListComponent implements OnInit {
+  ducks$!: Observable<Duck[]>;
+  @Input() duckIds: number[] = [];
+  @Input() isSelf: boolean = false;
+
+
+  constructor(private duckService: DuckService) { }
+
+  ngOnInit(): void {
+    this.ducks$ = this.duckService.get_ducks().pipe(
+      map( ducks => ducks.map(duck => {
+        duck.hidden = !this.duckIds.includes(duck.id);
```

Comment:
We shouldn't be updating the ducks in the service based on the ids in the component page. Why not have the duck service know about which ducks the user has and just mark the ones they don't have as hidden when downloading the list from the server? And then when awarding ducks, update that value?

It can just inject account.service to get that info.

Reply:
hmm true.

Reply:
but what about other people's ducks? imean i guess i can still just do it through the ducks endpoint...